KeyedDecodingContainer.swift

```diff
@@ -8,291 +8,214 @@
-                    value = Double(stringValue)
-                }catch{
-                }
-            }
+        if let doubleValue = try? decode(Double.self, forKey: key) {
+            return doubleValue
+        } else if let boolValue = try? decode(Bool.self, forKey: key) {
+            return boolValue ? 1 : 0
+        } else if let stringValue = try? decode(String.self, forKey: key) {
+            return Double(stringValue)
+        } else {
+            return 0.0
         }
-        return value
     }
     
-    //MARK: - URL
+    // MARK: - URL
+    
     func decodeIfPresent(_ type: URL.Type, forKey key: K) throws -> URL? {
-        var value:URL?
-        do {
-            value = try self.decode(URL.self, forKey: key)
-        }catch{
-        }
-        return value
+        return try? decode(URL.self, forKey: key)
     }
     
-    //MARK: - Date
+    // MARK: - Date
+    
     func decodeIfPresent(_ type: Date.Type, forKey key: K) throws -> Date? {
-        var value:Date?
-        do {
-            let timeStamp = try self.decode(Double.self, forKey: key)
+        if let timeStampValue = try? decode(Double.self, forKey: key) {
             if let div = Double(key.description) {
-                value = Date(timeIntervalSince1970: TimeInterval(timeStamp / div))
+                return Date(timeIntervalSince1970: TimeInterval(timeStampValue / div))
             }else {
-                value = Date(timeIntervalSince1970: TimeInterval(timeStamp))
-            }
-        }catch{
-            do {
-                value = try self.decode(Date.self, forKey: key)
-            }catch{
-                do {
-                    let stringValue = try self.decode(String.self, forKey: key)
-                    guard !key.description.isEmpty else {
-                        return value
-                    }
-                    value = self.fromStringToDate(dateString: stringValue, format: key.description)
-                } catch {
-                }
+                return Date(timeIntervalSince1970: TimeInterval(timeStampValue))
             }
+        } else if let dateValue = try? decode(Date.self, forKey: key) {
+            return dateValue
+        } else if let stringValue = try? decode(String.self, forKey: key), !key.description.isEmpty {
+            return stringValue.date(format: key.description)
+        } else {
+            return Date()
         }
-        return value
-    }
-    
-    fileprivate func fromStringToDate(dateString: String, format: String) -> Date?{
-        let dateFormatter = DateFormatter()
-        dateFormatter.dateStyle = .none;
-        dateFormatter.timeStyle = .none;
-        dateFormatter.timeZone = TimeZone.autoupdatingCurrent;
-        dateFormatter.dateFormat = format;
-        
-        guard let date = dateFormatter.date(from: dateString) else {
-            return nil
-        }
-        return date
     }
     
```

Comment:
@AnasAlhasani  Decodable Property removed from this pull request, we must have it in order to handle object property because we may receive String type and try to parse it as an object then we will receive mismatch type from JSONDecodable 😱